crates/wysiwyg/src/composer_model/replace_text.rs

```diff
@@ -67,16 +67,26 @@ where
         let leaves: Vec<&DomLocation> = range.leaves().collect();
         if leaves.len() == 1 {
             let location = leaves[0];
+            let current_cursor_global_location =
+                location.position + location.start_offset;
             let handle = &location.node_handle;
             let parent_list_item_handle =
                 self.state.dom.find_parent_list_item_or_self(handle);
             if let Some(parent_list_item_handle) = parent_list_item_handle {
+                let list_item_end_offset = range
+                    .locations
+                    .into_iter()
+                    .filter(|loc| {
+                        self.state.dom.lookup_node(&loc.node_handle).kind()
```

Comment:
I think you can just use `loc.kind() == DomNodeKind::ListItem` or `matches!(loc.kind(), ...)` for this now.